x-pack/plugins/security_solution/public/common/containers/sourcerer/index.tsx

```diff
@@ -180,28 +187,33 @@ export const useInitSourcerer = (
     },
     [defaultDataView.title, dispatch, indexFieldsSearch, addError]
   );
-  useEffect(() => {
+
+  const onSignalIndexUpdated = useCallback(() => {
     if (
       !loadingSignalIndex &&
       signalIndexName != null &&
       signalIndexNameSourcerer == null &&
       defaultDataView.id.length > 0
     ) {
-      // update signal name also updates sourcerer
-      // we hit this the first time signal index is created
       updateSourcererDataView(signalIndexName);
       dispatch(sourcererActions.setSignalIndexName({ signalIndexName }));
     }
   }, [
-    defaultDataView.id,
+    defaultDataView.id.length,
     dispatch,
-    indexFieldsSearch,
-    isSignalIndexExists,
     loadingSignalIndex,
     signalIndexName,
     signalIndexNameSourcerer,
     updateSourcererDataView,
   ]);
+
+  useEffect(() => {
+    onSignalIndexUpdated();
+    // because we only want onSignalIndexUpdated to run when signalIndexName updates,
+    // but we want to know about the updates from the dependencies of onSignalIndexUpdated
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [signalIndexName]);
```

Comment:
this was getting called twice, once when `signalIndexName` has a value and once when `signalIndexNameSourcerer` has a value. we only need to call it when `signalIndexName`